bundles/org.openhab.binding.enturno/src/main/java/org/openhab/binding/enturno/internal/connection/EnturNoConnection.java

```diff
@@ -0,0 +1,241 @@
+        }
+
+        Map<String, String> params = getRequestParams(handler.getEnturNoConfiguration());
+
+        EnturJsonData enturJsonData = gson.fromJson(getResponse(REALTIME_URL, params), EnturJsonData.class);
+
+        if (enturJsonData == null) {
+            throw new EnturCommunicationException("Error when deserializing response to EnturJsonData.class",
+                    new NullPointerException());
+        }
+
+        return processData(enturJsonData.data.stopPlace, lineCode);
+    }
+
+    private Map<String, String> getRequestParams(EnturNoConfiguration config) {
+        Map<String, String> params = new HashMap<>();
+        params.put(PARAM_STOPID, StringUtils.trimToEmpty(config.getStopPlaceId()));
+        params.put(PARAM_START_DATE_TIME, StringUtils.trimToEmpty(LocalDateTime.now(ZoneId.of(TIME_ZONE)).toString()));
+
+        return params;
+    }
+
+    private String getResponse(String url, Map<String, String> params) {
+        try {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Entur request: URL = '{}', graphQL parameters -> startTime = '{}', stopId = '{}'",
+                        REALTIME_URL, params.get(PARAM_START_DATE_TIME), params.get(PARAM_STOPID));
+            }
+
+            Request request = httpClient.newRequest(url);
+            request.method(POST);
+            request.timeout(10, TimeUnit.SECONDS);
+            request.header(HttpHeader.CONTENT_TYPE, CONTENT_TYPE);
+            request.header(REQUIRED_CLIENT_NAME_HEADER, REQUIRED_CLIENT_NAME);
+            request.content(new StringContentProvider(getRequestBody(params)));
+
+            logger.trace("Request body: {}", getRequestBody(params));
+
+            ContentResponse contentResponse = request.send();
+
+            int httpStatus = contentResponse.getStatus();
+            String content = contentResponse.getContentAsString();
+            String errorMessage = StringUtils.EMPTY;
+            logger.trace("Entur response: status = {}, content = '{}'", httpStatus, content);
+            switch (httpStatus) {
+                case OK_200:
+                    return content;
+                case BAD_REQUEST_400:
+                case NOT_FOUND_404:
+                    errorMessage = getErrorMessage(content);
+                    logger.debug("Entur server responded with status code {}: {}", httpStatus, errorMessage);
+                    throw new EnturConfigurationException(errorMessage);
+                default:
+                    errorMessage = getErrorMessage(content);
+                    logger.debug("Entur server responded with status code {}: {}", httpStatus, errorMessage);
+                    throw new EnturCommunicationException(errorMessage);
+            }
+        } catch (ExecutionException e) {
+            String errorMessage = e.getLocalizedMessage();
+            logger.debug("Exception occurred during execution: {}", errorMessage, e);
```

Comment:
In this case, when exception is caught only thing status is updated by base class of thing handler (I cannot see that it is logged anywhere there, so I think it would be nice to have it in logs as well). I am leaving logging as it is.

Reply:
> In this case, when exception is caught only thing status is updated by base class of thing handler

Then that is the other 'bad' practice, so if you follow the guideline to log when you handle it and otherwise rethrow it every exception should end up in the log once.